tokio-postgres/src/config.rs

```diff
@@ -90,6 +91,17 @@ pub enum Host {
 ///     path to the directory containing Unix domain sockets. Otherwise, it is treated as a hostname. Multiple hosts
 ///     can be specified, separated by commas. Each host will be tried in turn when connecting. Required if connecting
 ///     with the `connect` method.
+/// * `hostaddr` - Numeric IP address of host to connect to. This should be in the standard IPv4 address format,
+///     e.g., 172.28.40.9. If your machine supports IPv6, you can also use those addresses.
+///     If this parameter is not specified, the value of `host` will be looked up to find the corresponding IP address,
+///     - or if host specifies an IP address, that value will be used directly.
+///     Using `hostaddr` allows the application to avoid a host name look-up, which might be important in applications
+///     with time constraints. However, a host name is required for verify-full SSL certificate verification.
+///     Note that `host` is always required regardless of whether `hostaddr` is present.
```

Comment:
Why is this? psql at least will try to use just hostaddr if possible.

Reply:
Yeah I realized that too but making host optional would be a breaking change. Currently, in connect we iterate over the host vector. It'd be rather a big change. If you feel strongly about it I can try to fiddle with it.

Reply:
How would it be a breaking change? What code that used to work would now fail?

Reply:
Sure! Let me play with it. So the length of the two vectors host and hostaddr must be equal or one must be empty?

Reply:
Thanks! Let me know what you think in the latest commit https://github.com/sfackler/rust-postgres/pull/945/files. I added a few tests to demonstrate how it works.